Makefile.win

```diff
@@ -0,0 +1,233 @@
+-include Makefile.local # for optional local options e.g. threads
+
+# Recipes for this Makefile
+
+## Build the compiler
+##   $ make
+## Build the compiler with progress output
+##   $ make progress=true
+## Clean up built files then build the compiler
+##   $ make clean crystal
+## Build the compiler in release mode
+##   $ make crystal release=1
+## Run all specs in verbose mode
+##   $ make spec verbose=1
+
+CRYSTAL ?= crystal ## which previous crystal compiler use
+LLVM_CONFIG ?=     ## llvm-config command path to use
+
+release ?=      ## Compile in release mode
+stats ?=        ## Enable statistics output
+progress ?=     ## Enable progress output
+threads ?=      ## Maximum number of threads to use
+debug ?=        ## Add symbolic debug info
+verbose ?=      ## Run specs in verbose mode
+junit_output ?= ## Path to output junit results
+static ?=       ## Enable static linking
+interpreter ?=  ## Enable interpreter feature
+
+MAKEFLAGS += --no-builtin-rules
+.SUFFIXES:
+
+SHELL := cmd.exe
+CXX := cl.exe
+
+GLOB = $(shell dir $1 /B /S)
+MKDIR = if not exist $1 mkdir $1
+CP = copy /B /Y $1 $2
+CPDIR = robocopy /E /NJH /NJS $1 $2 & if %%ERRORLEVEL%% GEQ 8 exit /B 1
+INSTALL = copy /B /Y $1 $2
+INSTALLDIR = robocopy /E /NJH /NJS $1 $2 & if %%ERRORLEVEL%% GEQ 8 exit /B 1
+MV = move /Y $1 $2
+RM = if exist $1 del /F /Q $1
+RMDIR = if exist $1 rd /S /Q $1
+
+O := .build
+SOURCES := $(call GLOB,src\\*.cr)
+SPEC_SOURCES := $(call GLOB,spec\\*.cr)
+override FLAGS += -D strict_multi_assign $(if $(release),--release )$(if $(stats),--stats )$(if $(progress),--progress )$(if $(threads),--threads $(threads) )$(if $(debug),-d )$(if $(static),--static )$(if $(LDFLAGS),--link-flags="$(LDFLAGS)" )$(if $(target),--cross-compile --target $(target) )$(if $(interpreter),,-Dwithout_interpreter )
+SPEC_WARNINGS_OFF := --exclude-warnings spec\std --exclude-warnings spec\compiler --exclude-warnings spec\primitives
+SPEC_FLAGS := $(if $(verbose),-v )$(if $(junit_output),--junit_output $(junit_output) )
+CRYSTAL_CONFIG_LIBRARY_PATH := $$ORIGIN\lib
+CRYSTAL_CONFIG_BUILD_COMMIT := $(shell git rev-parse --short HEAD)
+CRYSTAL_CONFIG_PATH := $$ORIGIN\src
+SOURCE_DATE_EPOCH := $(shell git show -s --format=%ct HEAD)
+export_vars = $(eval export CRYSTAL_CONFIG_BUILD_COMMIT CRYSTAL_CONFIG_PATH SOURCE_DATE_EPOCH)
+export_build_vars = $(eval export CRYSTAL_CONFIG_LIBRARY_PATH)
+LLVM_CONFIG ?=
+LLVM_VERSION := $(if $(LLVM_CONFIG),$(shell $(LLVM_CONFIG) --version))
+LLVM_EXT_DIR = src\llvm\ext
+LLVM_EXT_OBJ = $(LLVM_EXT_DIR)\llvm_ext.obj
+DEPS = $(LLVM_EXT_OBJ)
+CXXFLAGS += $(if $(debug),/MTd /Od,/MT)
+CRYSTAL_VERSION ?= $(shell type src\VERSION)
+
+DESTDIR ?=
+prefix ?= $(if $(ProgramW6432),$(ProgramW6432),$(ProgramFiles))\crystal

```

Comment:
Can we swap the values of `DESTDIR` and `prefix`? The former is which you should usually override if you want to specify a custom install path. That wouldn't work if `prefix` is already anchored.
Perhaps we could also consider splitting the anchor from `ProgramFiles`/`ProgramW6432` using that as `DESTDIR` and the rest as `prefix` but I don't think there's much point in it. `prefix` has a specific purpose on unixoid systems which doesn't properly translate to Windows.

Reply:
No, this is still the correct use of `prefix`; `DESTDIR` is only used if `make install` places files in a location that isn't the final install directory. But you are probably right that `DESTDIR` as it stands now will fail if specified alone.

Reply:
What I think should happen is `DESTDIR`'s drive takes precedence over `prefix`'s; so if `DESTDIR=D:\Temp` and `prefix=C:\Users\Public`, then the compiler will be found at `D:\Temp\Users\Public\crystal.exe`. (If `$ORIGIN` in properties like `CRYSTAL_PATH` were to be replaced by a hard-coded path, it would be `$(prefix)` rather than `$(DESTDIR)$(prefix)`. This is what distinguishes the two variables when deciding which one to use.)

Reply:
I'm not very happy about the added complexity for this, which AFAIK doesn't even have a strong use case.

Reply:
`DESTDIR` is no more. Fortunately this doesn't matter for us because our build artifacts are truly portable (mostly due to compiler support for `$ORIGIN`) and staged installs are unnecessary.